analyzers/src/SonarAnalyzer.Common/Helpers/MemberDescriptor.cs

```diff
@@ -64,7 +64,7 @@ private bool HasSameContainingType<TSymbolType>(TSymbolType memberSymbol, bool c
             where TSymbolType : class, ISymbol
         {
             var containingType = memberSymbol.ContainingType?.ConstructedFrom;
-            return checkOverriddenMethods ? containingType.DerivesOrImplements(ContainingType) : containingType.Is(ContainingType);
+            return containingType != null && checkOverriddenMethods ? containingType.DerivesOrImplements(ContainingType) : containingType.Is(ContainingType);
```

Comment:
there are null checks inside `DerivesOrImplements` and `Is`, but this is better for readability

Reply:
Since **the method is private** we can accept that the parameter sanitization is done by the caller. From my point of view this is fine anyway.